lang/machine.md

```diff
@@ -57,30 +60,58 @@ Next, we define how to create a machine.
 
 ```rust
 impl<M: Memory> Machine<M> {
-    pub fn new(prog: Program) -> Option<Machine<M>> {
+    pub fn new(prog: Program) -> Option<NdResult<Machine<M>>> {
```

Comment:
This is a strange return type. If we truly need an NdResult here we probably want to also represent check_wf failure with an Err, not a None.